### PR TITLE
[Sprint 4] Feature Logic Request And Cancel Shop In Domain

### PR DESCRIPTION
**Pull request by tantuan127**
Feature Logic Request And Cancel Shop In Domain
Link redmine : https://dev.framgia.com/redmine/issues/116010
Task : Join Domain Shop Management
Effect: N/A

### CODE REVIEW

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java [hunk 1]
```diff
@@ -102,6 +122,27 @@ private void onCancel() {
...
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        dialog.cancel();
+                    }
+                })
+                .setPositiveButton(R.string.action_yes, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        mShopDomainManagementListener.onCancleJoinDomain();
+                        onRequest();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java]: why ``onRequest();`` call in here?

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java [hunk 2]
```diff
@@ -102,6 +122,27 @@ private void onCancel() {
         mTextButton.set(R.string.cancel);
     }
 
+    private void onDialog() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java]: Please declare in DialogManager

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java (hunk 1 above)
Comment (tantuan127) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java]: Dạ vì khi huỷ đăng ký thì button sẽ chuyển về trạng thái chưa đăng ký anh!

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java [hunk 3]
```diff
@@ -102,6 +126,24 @@ private void onCancel() {
...
     }
 
+    private void onRequestShopInDomain() {
+        ShopInDomainRequest shopInDomainRequest = new ShopInDomainRequest();
+        shopInDomainRequest.setDomainId(mShopInfo.getDomainId());
+        shopInDomainRequest.setShopName(mShop.getName());
+        shopInDomainRequest.setNumberProduct(mShopInfo.getNumberProduct());
+        shopInDomainRequest.setDomainName(mShopInfo.getDomainName());
+        mShopDomainManagementListener.onRequestJoinDomain(shopInDomainRequest);
+        onCancel();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java]: Please remove ``onCancel()``, It only call when server response success

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java [hunk 4]
```diff
@@ -102,6 +126,24 @@ private void onCancel() {
...
+    }
+
+    private void onCancelShopInDomain() {
+        ShopInDomainRequest shopInDomainCancel = new ShopInDomainRequest();
+        shopInDomainCancel.setDomainId(mShopInfo.getDomainId());
+        shopInDomainCancel.setShopId(mShop.getId());
+        mShopDomainManagementListener.onCancleJoinDomain(shopInDomainCancel);
+        onRequest();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ItemShopInfoViewModel.java]: Please remove ``onRequest()``, It only call when server response success

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementContract.java [hunk 5]
```diff
@@ -18,11 +19,25 @@
         void onGetListShopManagementError(BaseException exception);
 
         void onGetListShopManagementSuccess(List<ShopManagement> shopManagements);
+
+        void onRequestShopInDomainSuccess();
+
+        void onRequestShopInDomainError();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementContract.java]: Please pass BaseException as a param of this method

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementContract.java [hunk 6]
```diff
@@ -18,11 +19,25 @@
...
+
+        void onCancelJoinDomainSuccess();
+
+        void onCancelJoinDomainError();
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementContract.java]: Please pass BaseException as a param of this method

File: app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementViewModel.java [hunk 7]
```diff
@@ -22,6 +24,11 @@
     public ShopManagementViewModel(Navigator navigator, ListShopManagementAdapter adapter) {
         mNavigator = navigator;
         mListShopManagementAdapter = adapter;
+        mListShopManagementAdapter.setShopDomainManagementListener(this);
+    }
+
+    public ShopManagementContract.Presenter getPresenter() {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/screen/shopmanagement/ShopManagementViewModel.java]: remove this method

File: app/src/main/java/com/framgia/forder/data/source/remote/ShopRemoteDataSource.java [hunk 8]
```diff
@@ -84,4 +86,14 @@ public ShopRemoteDataSource(FOrderApi fOrderApi) {
...
                 });
     }
+
+    @Override
+    public Observable<ShopManagementResponse> requestShopInDomain(ShopInDomainRequest request) {
+        return mFOrderApi.requestShopInDomain(request);
+    }
+
+    @Override
+    public Observable<ShopManagementResponse> cancelShopInDomain(ShopInDomainCancel cancel) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/source/remote/ShopRemoteDataSource.java]: Please rename ``cancel`` to ``request``

File: app/src/main/java/com/framgia/forder/data/source/ShopRepository.java [hunk 9]
```diff
@@ -28,4 +31,12 @@ public ShopRepository(ShopRemoteDataSource remoteDataSource) {
     public Observable<List<ShopManagement>> getListShopManagement(int userId) {
         return mRemoteDataSource.getListShopManagement(userId);
     }
+
+    public Observable<ShopManagementResponse> requestShopInDomain(ShopInDomainRequest request) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/source/ShopRepository.java]: requestApplyShopToDomain()
requestLeaveShopFromDomain()

File: app/src/main/java/com/framgia/forder/data/source/ShopRepository.java [hunk 10]
```diff
@@ -28,4 +31,14 @@ public ShopRepository(ShopRemoteDataSource remoteDataSource) {
     public Observable<List<ShopManagement>> getListShopManagement(int userId) {
         return mRemoteDataSource.getListShopManagement(userId);
     }
+
+    public Observable<ShopManagementResponse> requestApplyShopToDomain(
+            ShopInDomainApplyRequest apply) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/source/ShopRepository.java]: change ``ShopInDomainApplyRequest`` to ``ApplyShopToDomainRequest``
and declare of param ``apply``  to ``applyShopToDomainRequest``

File: app/src/main/java/com/framgia/forder/data/source/ShopRepository.java [hunk 11]
```diff
@@ -28,4 +31,14 @@ public ShopRepository(ShopRemoteDataSource remoteDataSource) {
...
+        return mRemoteDataSource.requestApplyShopToDomain(apply);
+    }
+
+    public Observable<ShopManagementResponse> requestLeaveShopFromDomain(
+            ShopInDomainLeaveRequest leave) {
```
Comment (daolq3012) [inline app/src/main/java/com/framgia/forder/data/source/ShopRepository.java]: like a bove

**Pull request merged**